uaclient/api/u/pro/detect/should_auto_attach/v1.py

```diff
@@ -0,0 +1,41 @@
+from uaclient import exceptions
+from uaclient.api.api import APIEndpoint
+from uaclient.api.data_types import AdditionalInfo
+from uaclient.apt import is_installed
+from uaclient.clouds.identity import cloud_instance_factory
+from uaclient.config import UAConfig
+from uaclient.data_types import BoolDataValue, DataObject, Field
+
+
+class ShouldAutoAttachResult(DataObject, AdditionalInfo):
+    fields = [
+        Field("should_auto_attach", BoolDataValue),
+    ]
+
+    def __init__(self, should_auto_attach: bool):
+        self.should_auto_attach = should_auto_attach
+
+
+def should_auto_attach_on_machine(cfg: UAConfig):
+    try:
+        cloud_instance_factory()
+    except (
+        exceptions.CloudFactoryNoCloudError,
+        exceptions.CloudFactoryUnsupportedCloudError,
+        exceptions.CloudFactoryNonViableCloudError,
```

Comment:
These all extend `CloudFactoryError`
```suggestion
        exceptions.CloudFactoryError,
```

Reply:
Good point